src/resolvers/teams.js

```diff
@@ -102,11 +102,11 @@ const teamsResolvers = {
       if (team === null) {
         throw new Error('Team Not Found');
       }
-      team.name = name;
-      team.organization = organization;
-      team.amountRaised = amountRaised;
-      team.itemsSold = itemsSold;
-      team.isArchived = isArchived;
+      team.name = name || team.name;
+      team.organization = organization || team.organization;
+      team.amountRaised = amountRaised || team.amountRaised;
+      team.itemsSold = itemsSold || team.itemsSold;
+      team.isArchived = isArchived || team.isArchived;
```

Comment:
there must be a better way to write this. maybe like this? you should check Sequelize docs though for the right syntax and this this

```
async updateTeam(root, { id, ...rest }) {
  ...
  const team = await models.Team.update(rest, {
        where: {
          id,
        },
        returning: true,
      });
```

I remember @tawnyzhao mentioning it might be a good idea to create more specific mutations though, like `updateTeamName` - I don't know what the best pattern is tbh, might be doing some research into GraphQL best practices

Reply:
From the [Apollo Blog](https://www.apollographql.com/blog/designing-graphql-mutations-e09de826ed97/)

> Specific mutations that correspond to semantic user actions are more powerful than general mutations.

So I've split the mutations into 2 different ones and changed the implementation to use models.updates.

Reply:
ok, yeah this looks nicer! I'm thinking if it makes sense to be _that_ specific though, cuz now the client is making 2 requests in parallel (so in a real app you would actually want to _bundle_ the client requests together).

You could either do that, or an easier server side change would probably be to group these two into one mutation `updateTeamNameParts` maybe? You could leverage GraphQL fragments too if u wanted, https://www.apollographql.com/docs/react/data/fragments/

Happy to discuss this further though before u change anything, with others as well — else you might change the code and then we realize we want something else 🤣